### PR TITLE
Add case to run migration during file transfer between blk and scsi d…

### PR DESCRIPTION
**Pull request by suqinhuang**
Transfer file between blk and scsi disks and run migration

Signed-off-by: Suqin Huang <shuang@redhat.com>

id: 1408749

### CODE REVIEW

**Comment by suqinhuang** [comment]
https://github.com/autotest/tp-qemu/pull/905

**Comment by suqinhuang** [comment]
PASS 1-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp

**Comment by aliang123** [comment]
****Guest:Win10********************
Test cmd:
  python ConfigTest.py --guestname=Win10 --platform=x86_64 --verbose=yes --testcase=migrate.with_file_transfer.between_blk_scsi.tcp
Test Result:
 ERROR 1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp -> ShellStatusError: Could not get exit status of command: 'echo list disk > disk &&echo select disk 1 >> disk && echo create partition primary >> disk && echo exit>> disk && diskpart /s disk&& del /f disk'    (output: '\nMicrosoft DiskPart version 10.0.14393.0\n\nCopyright (C) 1999-2013 Microsoft Corporation.\nOn computer: DESKTOP-F2K5PNQ\n\n  Disk ###  Status         Size     Free     Dyn  Gpt\n  --------  -------------  -------  -------  ---  ---\n  Disk 0    Online         1024 MB  1024 MB         \n  Disk 1    Online           30 GB      0 B         \n  Disk 2    Online         1024 MB  1024 MB         \n\nDisk 1 is now the selected disk.\n\n')


****Guest:RHEL.7.3*****************
Test cmd:
  python ConfigTest.py --guestname=RHEL.7.3 --platform=x86_64 --verbose=yes --testcase=migrate.with_file_transfer.between_blk_scsi.tcp
Test Result:
   PASS 1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/866/files

**Comment by suqinhuang** [comment]
@aliang123 thanks for your test, could you try with above patch applied

**Comment by aliang123** [comment]
I have tried, but other problem occure:
  "ERROR 05-smp_8.4096m.repeat5.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp -> ShellTimeoutError: Timeout expired while waiting for shell command to complete: 'copy D:\\ed_1024.avi F:\\ed_1024.avi /y' "

Test cmd:
  python ConfigTest.py --guestname=Win10 --platform=x86_64 --verbose=yes --testcase=migrate.with_file_transfer.between_blk_scsi.tcp --nrepeat=30
Test Result:
 # python scan_results.py 
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'01-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:14:56 2017-01-22 21:21:24 PASS 387.589283
'02-smp_8.4096m.repeat2.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:21:24 2017-01-22 21:27:57 PASS 392.739522
'03-smp_8.4096m.repeat3.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:27:57 2017-01-22 21:34:24 PASS 387.038676
'04-smp_8.4096m.repeat4.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:34:24 2017-01-22 21:41:10 PASS 405.684634
'05-smp_8.4096m.repeat5.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:41:10 2017-01-22 21:44:20 ERROR 190.07724
'06-smp_8.4096m.repeat6.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:44:20 2017-01-22 21:51:03 PASS 402.769044
'07-smp_8.4096m.repeat7.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:51:03 2017-01-22 21:57:18 PASS 375.322942
'08-smp_8.4096m.repeat8.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 21:57:18 2017-01-22 22:04:00 PASS 402.469903
'09-smp_8.4096m.repeat9.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:04:01 2017-01-22 22:10:40 PASS 399.432312
'10-smp_8.4096m.repeat10.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:10:40 2017-01-22 22:17:06 PASS 385.667939
'11-smp_8.4096m.repeat11.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:17:06 2017-01-22 22:23:39 PASS 392.743135
'12-smp_8.4096m.repeat12.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:23:39 2017-01-22 22:30:10 PASS 391.098682
'13-smp_8.4096m.repeat13.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:30:10 2017-01-22 22:36:41 PASS 390.871088
'14-smp_8.4096m.repeat14.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:36:41 2017-01-22 22:43:25 PASS 404.700164
'15-smp_8.4096m.repeat15.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:43:26 2017-01-22 22:49:59 PASS 393.655476
'16-smp_8.4096m.repeat16.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:49:59 2017-01-22 22:56:37 PASS 397.493469
'17-smp_8.4096m.repeat17.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 22:56:37 2017-01-22 23:03:06 PASS 389.447634
'18-smp_8.4096m.repeat18.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:03:06 2017-01-22 23:09:46 PASS 399.647044
'19-smp_8.4096m.repeat19.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:09:46 2017-01-22 23:16:30 PASS 403.886569
'20-smp_8.4096m.repeat20.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:16:30 2017-01-22 23:23:04 PASS 394.013005
'21-smp_8.4096m.repeat21.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:23:04 2017-01-22 23:29:35 PASS 391.108185
'22-smp_8.4096m.repeat22.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:29:35 2017-01-22 23:36:10 PASS 394.256102
'23-smp_8.4096m.repeat23.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:36:10 2017-01-22 23:42:48 PASS 398.031513
'24-smp_8.4096m.repeat24.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:42:48 2017-01-22 23:49:30 PASS 402.209037
'25-smp_8.4096m.repeat25.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:49:30 2017-01-22 23:56:03 PASS 392.510619
'26-smp_8.4096m.repeat26.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-22 23:56:03 2017-01-23 00:02:36 PASS 393.317048
'27-smp_8.4096m.repeat27.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-23 00:02:36 2017-01-23 00:09:19 PASS 402.934768
'28-smp_8.4096m.repeat28.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-23 00:09:19 2017-01-23 00:15:51 PASS 391.661164
'29-smp_8.4096m.repeat29.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-23 00:15:51 2017-01-23 00:22:16 PASS 384.897729
'30-smp_8.4096m.repeat30.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.migrate.with_file_transfer.between_blk_scsi.tcp' Finshed 2017-01-23 00:22:16 2017-01-23 00:37:38 PASS 922.349641

File: qemu/tests/migration_during_test.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def run(test, params, env):
+    """
+    KVM migration test:
+    1) Get a live VM and clone it.
+    2) Verify that the source VM supports migration.  If it does, proceed with
+            the test.
+    3) Run sub test
+    4) Send a migration command to the source VM and wait until it's finished.
+    5) Kill off the source VM.
```
Comment (xutian) [inline qemu/tests/migration_during_test.py]: do you mean kill source VM, right?

File: qemu/tests/migration_during_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    :param env: Dictionary with the test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    mig_cancel_delay = int(params.get("mig_cancel") == "yes") * 2
```
Comment (xutian) [inline qemu/tests/migration_during_test.py]: not a good idea to do this.

**Pull request closed without merging**